### DISCUSSION TITLE
Closing all windows

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [tauri-apps/tauri#4863] [Q&A] [answered]
I'm positive this is probably covered somewhere, but couldn't find it.

Lets say my app have a main window, and then other windows that were created as a webview and/or from a command.

What's the way to go so when I close the main window actually all the others windows that were created close too?

**Comment by FabianLars** [comment, marked as answer]
That depends on if you want to exit the whole process, or just close the other windows but keeping the event loop running (for example if you want to keep it running in the system tray).

For the first version you'd listen to the CloseRequested event, check if it's the main window and then use `exit` [js](https://tauri.app/v1/api/js/modules/process#exit) / [rust](https://docs.rs/tauri/latest/tauri/struct.AppHandle.html#method.exit) to close everything.

For the second version, you'd have to keep track of the opened windows (their labels) yourself and then close them manually.
